markdown.dtx

```diff
@@ -25871,35 +26274,128 @@ M.extensions.tex_math_dollars = function()
         return (starter * C(p * (p - ender)^0) * ender)
       end
 
-      local inlinemathtail  = B( parsers.any * parsers.nonspacechar
-                               + parsers.backslash * parsers.any)
-                            * parsers.dollar 
-                            * -#(parsers.digit)
+      local allowed_before_closing = B( parsers.backslash * parsers.any
+                                      + parsers.any * (parsers.nonspacechar - parsers.backslash))
+
+      -- tex_math_dollars
+      local dollar_math_content = parsers.backslash^-1
+                                * parsers.any
+                                - parsers.blankline^2
+                                - parsers.dollar
+
+      local inline_math_opening_dollars = parsers.dollar
+                                        * #(parsers.nonspacechar)
+
+      local inline_math_closing_dollars = allowed_before_closing
+                                        * parsers.dollar
+                                        * -#(parsers.digit)
+
+      local inline_math_dollars = between(C( dollar_math_content),
+                                          inline_math_opening_dollars,
+                                          inline_math_closing_dollars)
+
+      local display_math_opening_dollars  = parsers.dollar
+                                          * parsers.dollar
+
+      local display_math_closing_dollars  = parsers.dollar
+                                          * parsers.dollar
 
-      local inlinemath = between(C( parsers.backslash^-1 
-                                  * parsers.any
-                                  - parsers.blankline^2 
-                                  - parsers.dollar),
-                                 parsers.dollar * #(parsers.nonspacechar),
-                                 inlinemathtail)
+      local display_math_dollars = between(C( dollar_math_content),
+                                           display_math_opening_dollars,
+                                           display_math_closing_dollars)
 
-      local displaymathdelim  = parsers.dollar 
-                              * parsers.dollar
+      -- tex_math_?_backslash
+      local backslash_math_content  = parsers.any 
+                                    - parsers.blankline^2
 
-      local displaymath = between(C( parsers.backslash^-1 
-                                   * parsers.any 
-                                   - parsers.blankline^2 
-                                   - parsers.dollar),
-                                  displaymathdelim,
-                                  displaymathdelim)
+      -- tex_math_double_backslash
```

Comment:
Instead of an opaque comment `-- tex_math_double_backslash`, how about a sentence or two in natural language?

```suggestion
%    \end{macrocode}
% \begin{markdown}
%
% The following patterns implement the Pandoc `tex_math_double_backslash` syntax extension.
%
% \end{markdown}
%  \begin{macrocode}
```